helpers.go

```diff
@@ -50,13 +57,90 @@ func GetUniqueTableReferences(node Node) []string {
 	return tableNames
 }
 
+// ValidateTargetTables recursively validates all tables found in the node and return them.
+func ValidateTargetTables(node Node) ([]*ValidatedTable, error) {
+	if node == nil {
+		return []*ValidatedTable{}, nil
+	}
+
+	tables := map[string]struct{}{}
+	validTables := []*ValidatedTable{}
+	err := Walk(func(node Node) (bool, error) {
+		if table, ok := node.(*Table); ok && table != nil && table.IsTarget {
+			tables[table.String()] = struct{}{}
+			validTable, err := ValidateTargetTable(table)
+			if err != nil {
+				return true, fmt.Errorf("validate: %s", err)
+			}
+			validTables = append(validTables, validTable)
+		}
+		return false, nil
+	}, node)
+	if err != nil {
+		return []*ValidatedTable{}, fmt.Errorf("walk subtree: %s", err)
+	}
+
+	return validTables, nil
+}
+
+func ValidateTargetTable(table *Table) (*ValidatedTable, error) {
+	if !table.IsTarget {
+		return nil, fmt.Errorf("table is not target")
+	}
+	tableNameRegEx, err := regexp.Compile("^([A-Za-z]+[A-Za-z0-9_]*)*(_[0-9]+){1,2}$")
+	if err != nil {
+		return nil, fmt.Errorf("regexp compile: %s", err)
+	}
+	if !tableNameRegEx.MatchString(table.String()) {
+		return nil, &ErrTableNameWrongFormat{Name: table.String()}
+	}
+
+	parts := strings.Split(table.String(), "_")
+	if len(parts) < 2 {
+		return nil, fmt.Errorf("chain id not present in table name")
+	}
```

Comment:
There's something weird here regarding that `{1, 2}`.
For read-query statements, there should be exactly 2. The `1` case sounds to me for CREATE TABLES. But that doesn't seem to be the intention because there's a `len(parts) < 2` below. So that if mostly means that we should expect exactly `2` and not `{1, 2}`?

Also, if I put `select * from hello_world_1` this would:
1) [Match the regex](https://regexr.com/73tn2) 
2) Also pass the check of `len(parts) < 2`

But probably, this wasn't the intention, since that SELECT table name isn't correct?
Let me know if I'm confused here.

Reply:
This is a good comment. I was more worried about checking if the name of the table is valid, regarding the type of the statement. 

With that perspective, `hello_world_1` would be a valid table name. But actually, there's a bug in [L112](https://github.com/tablelandnetwork/go-sqlparser/pull/36/files#diff-296a01a801f363652ec765418086e861f5db0287df573d54c8fde4eebe0e26e9R112). `parts[len(parts)-2] != ""` is checking only if the second last part is empty. It should check if it is a number. The fix would be:

```
_, err = strconv.ParseInt(parts[len(parts)-2], 10, 64)
if len(parts) == 2 || err != nil {
	parts = append(parts, "") // adds an empty tokenID in the end
}
```

**Additional info.** 
We are not currently validating the read table's name. [Example](https://testnet.tableland.network/query?s=select%20*%20from%20hello_world_1).

If the client wants to validate it, it could do:
```
validTable, _ := ValidateTargetTable(table)
if validTable.TokenID() == -1 {
  // throws err
}
```

Reply:
Well, I think you're making my point. The `ValidateTargetTable(...)` in my example won't fail, but the client should do an extra check to know that it is invalid, depending on the case. This situation of "not failing" on a method named "Validate..." but expecting the client to do something extra in some cases is what felt weird.

I can bet that most (if not all) clients trying to use this helper won't really know what they should do:
```
validTable, _ := ValidateTargetTable(table)
if validTable.TokenID() == -1 {
  // throws err
}
```

This public helper is pretty low level in my opinion. I don't think we can't do much about this apart from writing a comment in the method, since the parameter is a `Table` and lacks of context to improve the behavior.